### PR TITLE
Venter integration

### PR DESCRIPTION
**Pull request by anondita**

### CODE REVIEW

File: app/graphql/queries/consultation_response/venter_map.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module Queries
+	module ConsultationResponse
+		class VenterMap < Queries::BaseQuery
+	    description "Get resonse keyword map for consultation responses"
+	    type GraphQL::Types::JSON, null: false
+
+	    def resolve
+	    	hash_map_response = Rails.cache.read("response_hash_map")
```
Comment (tambakoo) [inline app/graphql/queries/consultation_response/venter_map.rb]: So, what happens when let's say the cache is wiped out and no responses are added. The `read` method will keep returning `nil` and then the cache will have to be manually created.

File: app/graphql/queries/consultation_response/venter_map.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module Queries
+	module ConsultationResponse
+		class VenterMap < Queries::BaseQuery
+	    description "Get resonse keyword map for consultation responses"
```
Comment (tambakoo) [inline app/graphql/queries/consultation_response/venter_map.rb]: typo :p .  I use sublime spell check

File: app/jobs/fetch_response_map_job.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+class FetchResponseMapJob < ApplicationJob
+  queue_as :default
+  require 'uri'
+	require 'net/http'
+
+  def perform(consultation_response)
+  	cr = consultation_response
+  	url = "http://35.223.43.106/venter/modelKM"
```
Comment (tambakoo) [inline app/jobs/fetch_response_map_job.rb]: 🚨 @anondita civis is open sourced. Don't expose the IP here. Also squash this commit please. 🚨

File: app/graphql/queries/consultation_response/venter_map.rb (hunk 2 above)
Comment (anondita) [reply app/graphql/queries/consultation_response/venter_map.rb]: done

File: app/graphql/queries/consultation/analysis_map.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+module Queries
+	module Consultation
+		class AnalysisMap < Queries::BaseQuery
```
Comment (mikevic) [inline app/graphql/queries/consultation/analysis_map.rb]:
```suggestion
		class Analysis < Queries::BaseQuery
```

File: app/graphql/queries/consultation/analysis_map.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+	    description "Get response keyword map for consultation responses"
```
Comment (mikevic) [inline app/graphql/queries/consultation/analysis_map.rb]:
```suggestion
	    description "Keyword analysis based on consultation responses"
```

File: app/graphql/queries/consultation/analysis_map.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+	    argument :id,		Int,		required: true
+	    
+	    type GraphQL::Types::JSON, null: false
+
+	    def resolve(id:)
+	    	hash_map_response = Rails.cache.read("response_hash_map_#{id}")
```
Comment (mikevic) [inline app/graphql/queries/consultation/analysis_map.rb]:
```suggestion
	    	consultation_keyword_analysis_hash = Rails.cache.read("consultation_keyword_analysis_#{id}")
```

File: app/jobs/fetch_response_map_job.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+class FetchResponseMapJob < ApplicationJob
```
Comment (mikevic) [inline app/jobs/fetch_response_map_job.rb]:
```suggestion
class AnalyseKeywordsForConsultationJob < ApplicationJob
```

File: config/credentials/development.yml.enc [hunk 8]
```diff
@@ -1 +1 @@
...
...od60HQpCQYg8gIXfrqhOKJUcQYcAvgNP2+tZtROcKJktj/dmmqdhM2zJAQBSTLrz65Aao9M7emrfu5GcNwK1Bz9X9sHxBI9SYVFo1od7w3Zvg3YfdDIH0xqsfXBSbw6dgjQEwaQ8bjIpNyKtwrsSS2yGjCUbTdK9P12yMRbZ3c2raIkfHxnMPBdGyS8/GHnfZNxOr5gAFQ8anoeB/7m+BmGAiiZmYSr0R/Pwzfrc8emlOc2SKo3+FgLPgCTdyeF9OxfPO+70zniDr50VG0MQEBS9ZZgHfQHwSj5idLNTPuSeInp83bAN4+nalsmmXk2hm8t2aMTsWH1QajDsSbstiaF2ptPua9DnTGi5KJcFTsfP3078jVFwdShYZo68cttSNs1O6klJRwPDm4GFuWYETwX5OqEklbDRP0SaDmmFtO5XUhYzHrYfMkBxkG8hDEQWHrYbamBvDBc6plDILkO+ULqY1qjylMIhNM0IN9AgJcWCtUZu7x2OJa57IxvCgRihwH54V8tASwWqnP8/ra1oOIPLx4mSzdNRSR2l54ORP6EMpsNu9pOi0dvGLHxXovnkAU8jNe5ArypZxPpS+chdMxm9T4rmp9Y+YaVMLRIoxXY6YkXFai5YC+Ci0Kef2nmch2BC7vt3JI++5uwoQwkVunJEjlQ5UIGW4X+L2ffmUN/BV7l4L/KWROVivoENG6itpwDJ8Zz2YI7P0jWACApb6x2pvJrjI9yxvrYFmkMR5WfIoZGfbjoULlyVofJFqhVHZCJF+m8gBLcwzTlopTgPvlwOMIgb4n6h07vuaEbAsEIMP0ejStOhdypd98vfpiPVInypYaAHg2WJgOkgbhFOJhw/O73YhaMqWVQJy//+/PmrE3N0NhP2jXyrmk+wxTq331Xq+UCnvRTQ7C29RqqfDVyux0vL3WYpL2zpm2yVLFZFzbhlZfFV6UAqBqZ3UzAGPKeMXIRF9zJzgR6vjtCY2PpbUArL8C4b/KxIlRZTbBGcBG0Y8qMg2deHGD6R9W4uDWmHgrLGD/kMhI5xnJ52YZAMzDhBmf+KxjAjCKxRLyQbTov730uZaeHw2m9EsifycmxKLmT4w/Dcq+sCOlItn8JVp9e3zya+VUttGU305jzvRVTc/orp6o7FKeciHjFsRu/n--zc4Iy8TT2gL+ybLn--9YJt69NVKAsh795+UE7ucQ==
```
Comment (mikevic) [inline config/credentials/development.yml.enc]: @anondita - please add the same URI for staging

File: app/graphql/types/query_type.rb [hunk 9]
```diff
@@ -7,6 +7,7 @@ class QueryType < Types::BaseObject
     field :case_study_profile,                 resolver: Queries::CaseStudy::Profile
   	field :constant_for_type,			             resolver: Queries::Constant::ForType
   	field :consultation_list,                  resolver: Queries::Consultation::List
+    field :consultation_analysis_map,          resolver: Queries::Consultation::Analysis
```
Comment (mikevic) [inline app/graphql/types/query_type.rb]:
```suggestion
    field :consultation_analysis,          resolver: Queries::Consultation::Analysis
```

File: app/models/consultation_response.rb [hunk 10]
```diff
@@ -11,6 +11,7 @@ class ConsultationResponse < ApplicationRecord
   has_many :down_votes, -> { down }, class_name: 'ConsultationResponseVote'
   has_many :votes, class_name: 'ConsultationResponseVote'
   before_commit :update_reading_time
+  after_create :fetch_response_map
```
Comment (mikevic) [inline app/models/consultation_response.rb]:
```suggestion
  after_create :analyse_response
```

**Pull request merged**